lib/Elastica/Query.php

```diff
@@ -100,15 +105,15 @@ public function setRawQuery(array $query)
      *
      * @return $this
      */
-    public function setQuery(AbstractQuery $query)
+    public function setQuery(AbstractQuery $query): self
     {
         return $this->setParam('query', $query);
     }
 
     /**
      * Gets the query object.
      *
-     * @return \Elastica\Query\AbstractQuery
+     * @return array|\Elastica\Query\AbstractQuery
```

Comment:
Where is this change coming from?

Reply:
Both `Query::create` and `Query::__construct` allow passing in an array as argument, which don't parse / convert this array into an ensemble of objects but keep it as-is, essentially leveraging the `setRawQuery` method.

Calling `getQuery` after taking the route above (or by just setting a raw query on the Query object) will return an array. I think that's something one should be aware of.

FWIW, I do like the option to set the raw query, however I'd rather have it in a way where it's not mingled with the usual query abstraction but has it's own representation to have better control over it during it's lifecycle.